bigchaindb/backend/mongodb/query.py

```diff
@@ -153,13 +153,17 @@ def get_spent(conn, transaction_id, output):
     cursor = conn.run(
         conn.collection('bigchain').aggregate([
             {'$match': {
-                'block.transactions.inputs.fulfills.txid': transaction_id,
-                'block.transactions.inputs.fulfills.output': output
+                'block.transactions.inputs.fulfills': {
```

Comment:
Well, perhaps [`elemMatch`](https://docs.mongodb.com/manual/reference/operator/query/elemMatch/#op._S_elemMatch) is better then.